src/main/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactory.java

```diff
@@ -140,10 +129,14 @@ public GenRepoInfoFile genRepoInfoFile() {
     return getGcloud().newGenRepoInfo(newDefaultProcessHandler());
   }
 
+  private CloudSdk getCloudSdk() {
+    return defaultCloudSdk(mojo, new CloudSdkOperationsFactory());
```

Comment:
We prob don't want to create a new `CloudSdk` each this this is called right?

Reply:
yeah, I guess we could singleton it, but I think each Mojo already has it's own instance of this Factory anyway. So unless we rewrite this whole thing as some kind of Singleton service that instantiated once, it shouldn't matter.

Reply:
I was thinking that if `mojo.getCloudSdkHome` was null, it would be downloading the CloudSDK every time `getCloudSdk` was called (L143).

Reply:
Download should only run conditionally, this is handled further down in the code? Lemme find it.

Reply:
Ah CloudSdkDownloader only conditionally downloads it.

Reply:
I think there some space to clean up this whole interaction. will do in a followup.

Reply:
It looks like the checking on L157 still happens right?